### DISCUSSION TITLE
How to find out what layer an IntGrid cell is from

### DISCUSSION THREAD

**Discussion by DeanJGrey** [Trouv/bevy_ecs_ldtk#385] [Q&A] [answered]
Hi, everyone.

I've been reading the docs for a while, but I'm still having trouble trying to find out how to find out what layer an IntGridCell component is from. I have seen mentions of LayerMetadata and LayerInstance, however I'm not sure how to use those in a system, when trying to query the correct IntGridCells.

Could someone please point me in the right direction?

Thank you, in advance.

**Comment by PraxTube** [comment, marked as answer]
The hierarchy is pretty well explained [here in the docs](https://trouv.github.io/bevy_ecs_ldtk/v0.10.0/explanation/anatomy-of-the-world.html). You can just query for the parent and then get the layer metadata, that will most likely contain all the data you need.

```rust
fn debuggy(
    q_layers: Query<&LayerMetadata>,
    q_tiles: Query<(&Parent, &IntGridCell, &TileEnumTags), Added<IntGridCell>>,
) {
    for (parent, cell, enum_tags) in &q_tiles {
        if let Ok(layer) = q_layers.get(parent.get()) {
            info!("{:?}", layer);
        }
        info!(
            "cell: {}, tags: {:?}, uid: {:?}",
            cell.value, enum_tags.tags, enum_tags.source_enum_uid
        );
    }
}
```
